File: Programming Basics with Python/T04 - Conditional Statements (Advanced)/exercise/08. On Time for the Exam.py
exam_hour = int(input())
exam_minutes = int(input())
arrival_hour = int(input())
arrival_minutes = int(input())
exam_time_in_minutes = (exam_hour * 60) + exam_minutes
arrival_time_in_minutes = (arrival_hour * 60) + arrival_minutes
diff = abs(exam_time_in_minutes - arrival_time_in_minutes)
hour = diff // 60
minutes = diff % 60

if exam_time_in_minutes == arrival_time_in_minutes:
    print("On time")

elif exam_time_in_minutes > arrival_time_in_minutes:
    if diff <= 30:
        print("On time")
        print(f"{minutes} minutes before the start")
    elif 30 < diff < 60:
        print("Early")
        print(f"{minutes} minutes before the start")
    elif diff > 59:
        print("Early")
        print(f"{hour}:{minutes:02d} hours before the start")

elif arrival_time_in_minutes > exam_time_in_minutes:
    print("Late")
    if diff < 60:
        print(f"{minutes} minutes after the start")
    else:
        print(f"{hour}:{minutes:02d} hours after the start")

File: Programming Basics with Go/T04 - Conditional Statements (Advanced)/exercise/08. On Time for the Exam.go
package main

import "fmt"

func main () {

	var examHour, examMiuntes int
	fmt.Scanln(&examHour)
	fmt.Scanln(&examMiuntes)

	var arriveHour, arriveMinute int
	fmt.Scanln(&arriveHour)
	fmt.Scanln(&arriveMinute)

	//време на изпит (час и минути) -> минути
	var examInMinutes int = examHour * 60 + examMiuntes
	//време на пристигане (час и минути) -> минути
	var arriveInMinutes int = arriveHour * 60 + arriveMinute

	if arriveInMinutes > examInMinutes {
		//закъсняваме
		fmt.Println("Late")
		//закъснението
		var late int = arriveInMinutes - examInMinutes
		if late < 60 {
			fmt.Printf("%d minutes after the start", late)
		} else {
			//закъснението >= 60 мин
			var lateHour int = late / 60
			var lateMinutes int = late % 60
			fmt.Printf("%d:%02d hours after the start", lateHour, lateMinutes)
		}
	} else if arriveInMinutes == examInMinutes || examInMinutes - arriveInMinutes <= 30 {
		//на време
		fmt.Println("On time")
		//на време с по-малко от 30 мин
		if examInMinutes - arriveInMinutes <= 30 {
			fmt.Printf("%d minutes before the start", examInMinutes - arriveInMinutes)
		}
	} else if examInMinutes - arriveInMinutes > 30 {
		//по-рано
		fmt.Println("Early")
		//подраняване
		var early int = examInMinutes - arriveInMinutes
		if early < 60 {
			fmt.Printf("%d minutes before the start", early)
		} else {
			//подраняване >= 60 мин
			var earlyHour int = early / 60
			var earlyMinutes int = early % 60
			fmt.Printf("%d:%02d hours before the start", earlyHour, earlyMinutes)
		}
	}



}